timestream_maker/src/astrometry_fcts.py

```diff
@@ -0,0 +1,211 @@
+    """ 
+    HArad = HA * np.pi/12; dec = np.radians(dec) ; lat = np.radians(lat)
+    za = np.arccos(np.sin(lat) * np.sin(dec) + np.cos(lat) * np.cos(dec) * np.cos(HArad))
+    return za
+
+def elevationAngle(dec,lat,HA): 
+    """
+    elevation angle (rad)
+
+    Parameters
+    ----------
+    dec: float 
+        declination angle in degrees     
+    lat: float
+        lattitude angle in degrees
+    HA: array
+        hour angle in hour
+
+    Returns
+    -------
+    ea: array
+        elevation angle in degree
+    """ 
+
+    return np.pi/2 - zenithAngle(dec,lat,HA)
+
+def azimuthAngle(dec,lat,HA):
+    """
+    source azimuth angle (rad)
+
+    Parameters
+    ----------
+    dec: float 
+        declination angle in degrees     
+    lat: float
+        lattitude angle in degrees
+    HA: array
+        hour angle in hour
+
+    Returns
+    -------
+    aa: array
+        source azimuth angle (rad)
+    """ 
+
+    za = zenithAngle(dec,lat,HA)
+    HArad = HA * np.pi/12; dec = np.radians(dec) ; lat = np.radians(lat)
+    cosAz = (np.sin(dec) - np.sin(lat) * np.cos(za))/(np.cos(lat) * np.sin(za))
+    sinAz = np.sin(HArad) * np.cos(dec) / np.sin(za)
+    return np.arctan2(sinAz,cosAz)
+
+def parallacticAngle(dec,lat,HA,unwrapPA=True):
+    """
+    source parrallactic angle (rad)
+
+    Parameters
+    ----------
+    dec: float 
+        declination angle in degrees     
+    lat: float
+        lattitude angle in degrees
+    HA: array
+        hour angle in degree
+
+    Returns
+    -------
+    pa: array
+        source parrallactic angle (rad)
+    """ 
+    HArad = HA * np.pi/12; dec = np.radians(dec) ; lat = np.radians(lat)
+    pa = np.arctan2(np.sin(HArad), np.cos(dec) * np.tan(lat) - np.sin(dec) * np.cos(HArad))
+    if unwrapPA == True:
+        pa = np.unwrap(pa)
+        if np.mean(pa)<0:
+            pa = pa + 2*np.pi
+    return pa
+
+def declinationAngle(azi, alt, lat):
+    """
+    source declination angle (rad)
+
+    Parameters
+    ----------
+    azi: float 
+        azimuth in degrees     
+    alt: float
+        lattitude angle in degrees
+    lat: float
+        lattitude angle in degree
+
+    Returns
+    -------
+    Dec: float
+        source declination angle (rad)
+    """ 
+
+    azi = np.radians(azi); alt = np.radians(alt); lat = np.radians(lat)
+    sinDec = np.sin(alt)*np.sin(lat) + np.cos(alt)*np.cos(lat)*np.cos(azi)
+    return np.arcsin(sinDec)
+
+def hourAngle(azi, alt, lat):
+    """
+    source hour angle (rad)
+
+    Parameters
+    ----------
+    azi: float 
+        azimuth in degrees     
+    alt: float
+        lattitude angle in degrees
+    lat: float
+        lattitude angle in degree
+
+    Returns
+    -------
+    ha: float
+        source hour angle (rad)
+    """ 
+    dec = declinationAngle(azi, alt, lat)
+    azi = np.radians(azi); alt = np.radians(alt); lat = np.radians(lat)
+    cosHA = ( np.sin(alt)- np.sin(dec)*np.sin(lat) )/ (np.cos(dec)*np.cos(lat))
```

Comment:
can we swap this to the tangent of the HA? I can't find any reference for this that would imply this is 100% correct. see: [here](https://astrogreg.com/snippets/altaztoHAdec.html)